File: app/src/main/java/com/akvelon/myfirstapp/lesson7/data/network/clients/RxWeatherClient.java
package com.akvelon.myfirstapp.lesson7.data.network.clients;

import com.akvelon.myfirstapp.lesson4.rxweather.data.models.DayForecastResponse;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface RxWeatherClient {

    @GET("/data/2.5/weather")
    Single<DayForecastResponse> getDailyForecastByCity(@Query("q") String cityName);

}

File: app/src/main/java/com/akvelon/myfirstapp/lesson9/data/network/clients/RxWeatherClient.kt
package com.akvelon.myfirstapp.lesson9.data.network.clients

import com.akvelon.myfirstapp.lesson9.data.models.DayForecastResponse
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface RxWeatherClient {

    @GET("/data/2.5/weather")
    fun getDailyForecastByCity(@Query("q") cityName: String): Single<DayForecastResponse>
}